proxy/src/serverless/conn_pool.rs

```diff
@@ -261,6 +265,23 @@ pub struct GlobalConnPoolOptions {
     pub max_total_conns: usize,
 }
 
+#[derive(Debug, thiserror::Error)]
```

Comment:
I would suggest to move it to `../backend.rs` and rename it to `HttpConnError` or something.

The idea is to separate the connection logic from pool.